### DISCUSSION TITLE
Key modifiers (e.g. shift, alt, ctrl) for other events than keyboard events

### DISCUSSION THREAD

**Discussion by pohlt** [alpinejs/alpine#4208] [3. Feature ideas]
I'd love to use modifiers such as `.shift`, `.alt`, or `.ctrl` for non-keyboard events such as mouse events: `@wheel.alt="zoom()"`

**Comment by ekwoka** [comment]
Well, mouse events don't have key modifiers.

**Comment by pohlt** [comment]
I think they do: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent

13 May 2024 18:16:48 Eric Kwoka ***@***.***>:

**Reply by ekwoka** [reply]
oh, wow, true!

yeah, key modifiers and button modifiers may make sense. 🤔 

https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent

PointerEvents don't, which is what Chrome and Safari send, but they may implement the full MouseEvent as well...

**Comment by ekwoka** [comment]
Handled in #4209

**Reply by pohlt** [reply]
Wow, that "escalated" quickly! 😉 

I didn't look into your PR, but did you consider mouse _wheel_ events as well?

Cheers, mate! 🍻

**Reply by ekwoka** [reply]
well, those are deprecated but yes, it works for those too.

**Reply by pohlt** [reply]
Are they? Where did you find that?

I read [this](https://developer.mozilla.org/en-US/docs/Web/API/Element/wheel_event) and don't see it mentioned.

**Reply by ekwoka** [reply]
https://developer.mozilla.org/en-US/docs/Web/API/Element/mousewheel_event

Ah, wheel, not mousewheel.

That is not included there. :thinking: should be simple I can add it...

**Reply by pohlt** [reply]
Great, the PR got merged. Thank you so much, @ekwoka!

Could you please do the same magic for the wheel events? That was my initial "itch" so to say.
